xradar/io/backends/nexrad_level2.py

```diff
@@ -0,0 +1,1381 @@
+            else:
+                cut_parameters = [{"elevation_angle": 0.0}] * self.nscans
+            scale = 360.0 / 65536.0
+            return np.array(
+                [cut_parameters[i]["elevation_angle"] * scale for i in scans],
+                dtype="float32",
+            )
+        else:
+            scale = 180 / (4096 * 8.0)
+            msgs = [self.radial_records[self.scan_msgs[i][0]] for i in scans]
+            return np.round(
+                np.array(
+                    [m["msg_header"]["elevation_angle"] * scale for m in msgs],
+                    dtype="float32",
+                ),
+                1,
+            )
+
+    def get_nyquist_vel(self, scans=None):
+        """
+        Retrieve the Nyquist velocities of the requested scans.
+
+        Parameters
+        ----------
+        scans : list or None
+            Scans (0 based) for which the Nyquist velocities will be
+            retrieved. None (the default) will return the velocities for all
+            scans in the volume.
+
+        Returns
+        -------
+        velocities : ndarray
+            Nyquist velocities (in m/s) for the requested scans.
+
+        """
+        if scans is None:
+            scans = range(self.nscans)
+        return self._radial_sub_array(scans, "nyquist_vel") * 0.01
+
+    def get_unambigous_range(self, scans=None):
+        """
+        Retrieve the unambiguous range of the requested scans.
+
+        Parameters
+        ----------
+        scans : list or None
+            Scans (0 based) for which the unambiguous range will be retrieved.
+            None (the default) will return the range for all scans in the
+            volume.
+
+        Returns
+        -------
+        unambiguous_range : ndarray
+            Unambiguous range (in meters) for the requested scans.
+
+        """
+        if scans is None:
+            scans = range(self.nscans)
+        # unambiguous range is stored in tenths of km, x100 for meters
+        return self._radial_sub_array(scans, "unambig_range") * 100.0
+
+    def get_data(self, moment, max_ngates, scans=None, raw_data=False):
+        """
+        Retrieve moment data for a given set of scans.
+
+        Masked points indicate that the data was not collected, below
+        threshold or is range folded.
+
+        Parameters
+        ----------
+        moment : 'REF', 'VEL', 'SW', 'ZDR', 'PHI', 'RHO', or 'CFP'
+            Moment for which to to retrieve data.
+        max_ngates : int
+            Maximum number of gates (bins) in any ray.
+            requested.
+        raw_data : bool
+            True to return the raw data, False to perform masking as well as
+            applying the appropiate scale and offset to the data.  When
+            raw_data is True values of 1 in the data likely indicate that
+            the gate was not present in the sweep, in some cases in will
+            indicate range folded data.
+        scans : list or None.
```

Comment:
This seems like we should be double careful here to not read two sweeps with different gate-spacing here. Do we have a test file which actually has different gate-spacing?